simapp/simd/cmd/root.go

```diff
@@ -50,10 +50,13 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 		WithHomeDir(simapp.DefaultNodeHome).
 		WithViper()
 
+	initClientCtx = clicfg.UpdateClientContextFromClientConfig(initClientCtx)
+
 	rootCmd := &cobra.Command{
 		Use:   "simd",
 		Short: "simulation app",
 		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
+
```

Comment:
I have also tried to put clicfg.UpdateClientContextFromClientConfig(initClientCtx)  inside PersistentPreRunE. In this case I can handle an error if  there is any error reading from client.toml file. I have got the same test results.